packages/twenty-front/src/modules/object-record/record-show/hooks/useRecordShowPagePagination.ts

```diff
@@ -0,0 +1,224 @@
+        (view: any) =>
+          view?.id === viewIdQueryParam &&
+          view?.objectMetadataId === objectMetadataItem.id,
+      ) as View;
+    }
+  }, [viewIdQueryParam, objectMetadataItem, views]);
+
+  const activeFieldMetadataItems = useMemo(
+    () =>
+      objectMetadataItem
+        ? objectMetadataItem.fields.filter(
+            ({ isActive, isSystem }) => isActive && !isSystem,
+          )
+        : [],
+    [objectMetadataItem],
+  );
+
+  const filterDefinitions = formatFieldMetadataItemsAsFilterDefinitions({
+    fields: activeFieldMetadataItems,
+  });
+
+  const sortDefinitions = formatFieldMetadataItemsAsSortDefinitions({
+    fields: activeFieldMetadataItems,
+  });
+
+  const filter = turnObjectDropdownFilterIntoQueryFilter(
+    mapViewFiltersToFilters(view?.viewFilters ?? [], filterDefinitions),
+    objectMetadataItem?.fields ?? [],
+  );
+
+  const orderBy = turnSortsIntoOrderBy(
+    objectMetadataItem,
+    mapViewSortsToSorts(view?.viewSorts ?? [], sortDefinitions),
+  );
+
+  const { fetchMoreRecords } = useFindManyRecords({
+    filter,
+    orderBy,
+    skip: isLoadedRecords,
+    objectNameSingular,
+    recordGqlFields: generateDepthOneRecordGqlFields({
+      objectMetadataItem,
+    }),
+    onCompleted: (records, options) => {
+      setObjectRecords(records);
+      setTotalRecords(options?.totalCount ?? 0);
+      setHasNextPage(options?.pageInfo?.hasNextPage);
+      setIsLoadingPagination(false);
+    },
+  });
+
+  useEffect(() => {
+    if (
+      !isLoadingPagination &&
+      objectRecords.length > 0 &&
+      isDefined(hasNextPage)
+    ) {
+      const recordIndex = objectRecords.findIndex(
+        (rec) => rec.id === objectRecordId,
+      );
+      if (recordIndex < 0 && hasNextPage) {
+        fetchMoreRecords();
+      } else if (recordIndex < 0 && !hasNextPage) {
+        throw new Error('Object name or Record id not found');
+      }
+      setHasPreviousRecord(recordIndex !== 0);
+      setHasNextRecord(recordIndex !== objectRecords.length || hasNextPage);
+      setCurrentRecordIndex(recordIndex);
+      if (recordIndex >= 0) setIsLoadedRecords(true);
+    }
+
+    if (isDefined(view)) {
+      setViewName(view.name);
+    }
+  }, [
+    objectRecordId,
+    isLoadingPagination,
+    objectRecords,
+    hasNextPage,
+    view,
+    fetchMoreRecords,
+    setHasPreviousRecord,
+    setHasNextRecord,
+    setCurrentRecordIndex,
+    setViewName,
+    setIsLoadedRecords,
+  ]);
+
+  useEffect(() => {
+    const previousIndex = currentRecordIndex - 1;
+    const nextIndex = currentRecordIndex + 1;
+    setHasPreviousRecord(previousIndex >= 0);
+    const nextRecordIsLast = nextIndex === objectRecords.length;
+
+    if (nextRecordIsLast === true && hasNextPage === true) {
+      setIsLoadedRecords(false);
+      fetchMoreRecords();
+    }
+    setHasNextRecord(nextIndex < objectRecords.length);
+  }, [
+    hasNextPage,
+    currentRecordIndex,
+    objectRecords,
```

Comment:
objectRecords feels to generic, if you fetch Ids only maybe you can also come up with a name that speaks more about what you're doing (e.g. orderedListOfObjectIds or something better you can come up with!)

Reply:
We need to fetch records